src/hotspot/share/runtime/lockStack.cpp

```diff
@@ -65,7 +65,7 @@ void LockStack::verify(const char* msg) const {
   assert((_top >= start_offset()), "lockstack underflow: _top %d end_offset %d", _top, start_offset());
   int top = to_index(_top);
   for (int i = 0; i < top; i++) {
-    assert(_base[i] != nullptr, "no zapped before top");
+    assert(_base[i] != nullptr || !is_owning_thread(), "no zapped before top");
     for (int j = i + 1; j < top; j++) {
       assert(_base[i] != _base[j], "entries must be unique: %s", msg);
```

Comment:
Okay so you tweaked the assert to allow a `nullptr` value when the caller
is not the owning thread. Got it.

Is it possible for `_base[i]` and `_base[j]` to both be `nullptr` when the
caller is not the owning thread? If so, then that assert will also fire...

Reply:
Aww right. The whole block is not safe to verify when not called from the owning thread, because the owning thread may modify everything under our feet. I've changed it so that the whole loops are only done when called from owning thread.